Docker registry for function-controller (#7244)
kyma-project/kyma: components/function-controller/README.md

| **CONTROLLER_DOCKER_REGISTRY_FQDN**             | function-controller-docker-registry.kyma-system.svc.cluster.local | Fully qualified domain name of Docker Registry service                                                                                                                                                                                     |
| **CONTROLLER_DOCKER_REGISTRY_PORT**             | 5000                                                              | Docker Registry service port through which communication is conducted                                                                                                                                                                      |
| **CONTROLLER_DOCKER_REGISTRY_EXTERNAL_ADDRESS** | https://registry.kyma.local                                       | Domain address through which connection to Docker Registry is facilitated                                                                                                                                                                  |

> | **CONTROLLER_IMAGE_PULL_SECRET_NAME**           | regcred                                                           | Name of the secret image pull secret which contains hashed credentials to Docker Registry                                                                                                                                                  |

Reviewer: ```suggestion
| **CONTROLLER_IMAGE_PULL_SECRET_NAME**           | `regcred`                                                           | Name of the image Pull Secret which contains hashed credentials to the Docker Registry                                                                                                                                                  |
```

> | **CONTROLLER_TEKTON_REQUESTS_CPU**              | 350m                                                              | Specifies minimal amount of CPU assigned to TaskRun building lambda image. Follow [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document to get to know available values    |

Reviewer: ```suggestion
| **CONTROLLER_TEKTON_REQUESTS_CPU**              | `350m`                                                              |  Minimum amount of CPU assigned to the TaskRun to build the lambda image. See [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document for available values.   |
```

Reviewer: I removed `Specifies` to be consistent with other descriptions in the table. I also changes `Follow` into `See `as I guess you rather follow some steps in instruction ...

> | **CONTROLLER_TEKTON_REQUESTS_MEMORY**           |                                                                   | Specifies minimal amount of Memory assigned to TaskRun building lambda image. Follow [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document to get to know available values |

Reviewer: ```suggestion
| **CONTROLLER_TEKTON_REQUESTS_MEMORY**           |                                                                   | Minimum amount of Memory assigned to the TaskRun to build the lambda image. See [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document for available values. |
```

> | **CONTROLLER_TEKTON_REQUESTS_MEMORY**           | Specifies minimal amount of Memory assigned to TaskRun building lambda image. Follow [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document to get to know available values | 600Mi                                                               |

Reviewer: ```suggestion
| **CONTROLLER_TEKTON_REQUESTS_MEMORY**           | Minimum amount of Memory assigned to the TaskRun to build the lambda image. See [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document for available values. | `600Mi`                                                               |
```

> | **CONTROLLER_TEKTON_LIMITS_CPU**                |                                                                   | Specifies maximum amount of CPU assigned to TaskRun building lambda image. Follow [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document to get to know available values    |

Reviewer: ```suggestion
| **CONTROLLER_TEKTON_LIMITS_CPU**                |                                                                   | Maximum amount of CPU assigned to the TaskRun to build the lambda image. See [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document for available values.  |
```

> | **CONTROLLER_TEKTON_LIMITS_CPU**                | Specifies maximum amount of CPU assigned to TaskRun building lambda image. Follow [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document to get to know available values    | 400m                                                                |

Reviewer: ```suggestion
| **CONTROLLER_TEKTON_LIMITS_CPU**                | Minimum amount of CPU assigned to the TaskRun to build the lambda image. See [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document for available values.  | `400m`                                                                |
```

> | **CONTROLLER_TEKTON_LIMITS_MEMORY**             |                                                                   | Specifies minimal amount of Memory assigned to TaskRun building lambda image. Follow [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document to get to know available values |

Reviewer: ```suggestion
| **CONTROLLER_TEKTON_LIMITS_MEMORY**             |                                                                   | Minimum amount of Memory assigned to the TaskRun to build the lambda image. See [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document for available values. |
```

> | **CONTROLLER_TEKTON_LIMITS_MEMORY**             | Specifies minimal amount of Memory assigned to TaskRun building lambda image. Follow [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document to get to know available values | 700Mi                                                               |

Reviewer: ```suggestion
| **CONTROLLER_TEKTON_LIMITS_MEMORY**             | Minimum amount of Memory assigned to the TaskRun to build the lambda image. See [this](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu) document for available values. | `700Mi`                                                              |
```

> > **NOTE:** Tekton handles resources in a specific way depending on Limitranges. You can find more about it [here](https://github.com/tektoncd/pipeline/blob/master/docs/taskruns.md#limitranges).

Reviewer: ```suggestion
> **NOTE:** Tekton handles resources in a specific way depending on limit ranges. You can find more about it [here](https://github.com/tektoncd/pipeline/blob/master/docs/taskruns.md#limitranges).
```

> > **CAUTION:** Functions works only in the `serverless` Namespace that is created as part of the `function-controller` chart. To avoid authentication issues, add a trusted certificate to your cluster. Read [here](https://kyma-project.io/docs/#installation-install-kyma-with-your-own-domain-generate-the-tls-certificate) how to obtain one.

Reviewer: ```suggestion
> **CAUTION:** Functions work only in the `serverless` Namespace that is created as part of the `function-controller` chart. To avoid authentication issues, add a trusted certificate to your cluster. Read [here](https://kyma-project.io/docs/#installation-install-kyma-with-your-own-domain-generate-the-tls-certificate) how to obtain one.
```

Reviewer: I would put it in the Overview, it would be easily seen then.
